giraph-core/src/main/java/org/apache/giraph/master/BspServiceMaster.java

```diff
@@ -1734,7 +1735,7 @@ private CheckpointStatus getCheckpointStatus(long superstep) {
     if (checkpointFrequency == 0) {
       return CheckpointStatus.NONE;
     }
-    long firstCheckpoint = INPUT_SUPERSTEP + 1 + checkpointFrequency;
+    long firstCheckpoint = INPUT_SUPERSTEP + 1;
```

Comment:
This will actually checkpoint before running superstep 0. You're basically checkpointing the input loading work.